bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java

```diff
@@ -495,30 +499,7 @@ private int getCurrentArrayLength() {
     public String stringValue() {
         if (elementType != null) {
             StringJoiner sj = new StringJoiner(" ");
-            if (elementType.getTag() == TypeTags.INT_TAG) {
-                for (int i = 0; i < size; i++) {
-                    sj.add(Long.toString(intValues[i]));
-                }
-                return sj.toString();
-            } else if (elementType.getTag() == TypeTags.BOOLEAN_TAG) {
-                for (int i = 0; i < size; i++) {
-                    sj.add(Boolean.toString(booleanValues[i]));
-                }
-                return sj.toString();
-            } else if (elementType.getTag() == TypeTags.BYTE_TAG) {
-                for (int i = 0; i < size; i++) {
-                    sj.add(Long.toString(Byte.toUnsignedLong(byteValues[i])));
-                }
-                return sj.toString();
-            } else if (elementType.getTag() == TypeTags.FLOAT_TAG) {
-                for (int i = 0; i < size; i++) {
-                    sj.add(Double.toString(floatValues[i]));
-                }
-                return sj.toString();
-            } else if (elementType.getTag() == TypeTags.STRING_TAG) {
-                for (int i = 0; i < size; i++) {
-                    sj.add(stringValues[i]);
-                }
+            if (generateStringForBasicTypes(sj)) {
```

Comment:
Shall we refactor this, the method looks like validation, but it does more than that.

Reply:
will refactor it